src/invision/components/movie-popup/movie.popup.js

```diff
@@ -1,89 +1,124 @@
 import React, { useState } from "react";
 import Modal from 'react-awesome-modal';
+import { useFormik } from 'formik';
 
 export default function MoviePopup(props) {
 
 
   const isAdd = props.title === 'Add Movie';
 
-  const [movie, setState] = useState({ ...props.movie });
+  const [isSubmitClicked, updateSubmitClick] = useState(false);
 
-  const reset = () => {
-    setState({
-      ...props.movie,
-    });
-  }
 
-  const handleTitleChange = (event) => {
-    setState({ ...movie, title: event.target.value });
-  }
 
-  const handleReleaseDateChange = (event) => {
-    setState({ ...movie, releasedDate: event.target.value });
-  }
+  const formik = useFormik({
+    initialValues: { ...props.movie },
+    validate: (values) => {
+      const errors = {};
 
-  const handleUrlChange = (event) => {
-    setState({ ...movie, thumbnail: event.target.value });
-  }
-  const handleGenreChange = (event) => {
-    setState({ ...movie, genre: event.target.value });
-  }
+      if (!values.title.trim()) {
+        errors.title = 'required';
+      }
 
-  const handleRuntimeChange = (event) => {
-    setState({ ...movie, runTime: event.target.value });
-  }
+      if (!values.releasedDate) {
+        errors.releasedDate = 'required';
+      }
 
-  const handleOverviewChange = (event) => {
-    setState({ ...movie, overview: event.target.value });
-  }
-  const handleRatingChange = (event) => {
-    setState({ ...movie, rating: event.target.value });
+      const urlRegex = /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+(?:png|jpg|jpeg|gif|svg)+$/g;
+      if (!values.thumbnail) {
+        errors.thumbnail = 'required';
+      } else if (!urlRegex.test(values.thumbnail)) {
+        errors.thumbnail = 'Url format expected';
+      }
+
+      if (!values.genre.trim()) {
+        errors.genre = 'required';
+      }
+
+      if (!values.overview) {
+        errors.overview = 'required';
+      } else if (values.overview.length < 5) {
+        errors.overview = 'Overview should be more than 5 chars';
+      }
+
+      if (!values.runTime) {
+        errors.runTime = 'required';
+      } else if (!isFloat(values.runTime)) {
+        errors.runTime = 'Runtime should be a number';
+      }
+
+      if (!values.rating) {
+        errors.rating = 'required';
+      } else if (!isFloat(values.rating)) {
```

Comment:
"vote_average" must be less than or equal to 100. I've gotten this error when I've tried to create movie with "2323" rating. You should add this rule to these validation list

Reply:
fixed